packages/playwright/src/reporters/html.ts

```diff
@@ -499,7 +499,7 @@ class HtmlBuilder {
 
   private _serializeAnnotations(annotations: api.TestCase['annotations']): TestAnnotation[] {
     // Annotations can be pushed directly, with a wrong type.
-    return annotations.map(a => ({ type: a.type, description: a.description === undefined ? undefined : String(a.description) }));
+    return annotations.map(a => ({ type: a.type, description: a.description === undefined ? undefined : String(a.description), location: a.location }));
```

Comment:
Let's expand all properties of `a.location` explicitly, similar to all other properties here.